libsolidity/Parser.cpp

```diff
@@ -792,7 +792,15 @@ ASTPointer<Expression> Parser::parseLeftHandSideExpression(
 			std::tie(arguments, names) = parseFunctionCallArguments();
 			nodeFactory.markEndPosition();
 			expectToken(Token::RParen);
-			expression = nodeFactory.createNode<FunctionCall>(expression, arguments, names);
+			if (Identifier* identifier = dynamic_cast<Identifier*>(expression.get()))
+			{
+				ASTPointer<FunctionIdentifier> functionIdentifier = make_shared<FunctionIdentifier>(identifier->getLocation(), make_shared<ASTString>(identifier->getName()));
+				auto functionCall = nodeFactory.createNode<FunctionCall>(functionIdentifier, arguments, names);
+				functionIdentifier->setFunctionCall(functionCall);
```

Comment:
Good catch! This piece of code doesn't smell well...

Another approach I'm thinking about is I don't create a new `FunctionIdentifier` node, just use this `Identifier` node and then store declarations into this node. Let me check which is better.